code/game/objects/items/weapons/RCD.dm

```diff
@@ -51,24 +29,36 @@ GLOBAL_LIST_INIT(rcd_door_types, list(
 	req_access = list(ACCESS_ENGINE)
 	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 50)
 	resistance_flags = FIRE_PROOF
-	// Important shit
+	/// The spark system used to create sparks when the user interacts with the RCD.
 	var/datum/effect_system/spark_spread/spark_system
-	var/matter = 0
-	var/max_matter = 100
-	var/mode = RCD_MODE_TURF
-	var/canRwall = 0
-
-	// UI shit
-	var/menu = RCD_PAGE_MAIN
+	/// The current amount of matter stored.
+	var/matter = NONE
+	/// The max amount of matter that can be stored.
+	var/max_matter = MATTER_100
+	/// The RCD's current build mode.
+	var/mode = MODE_TURF
+	/// If the RCD can deconstruct reinforced walls.
+	var/canRwall = FALSE
+	/// Is the RCD's airlock access selection menu locked?
 	var/locked = TRUE
+	/// The current airlock type that will be build.
 	var/door_type = /obj/machinery/door/airlock
+	/// The name that newly build airlocks will receive.
 	var/door_name = "Airlock"
-	var/list/door_accesses = list()
-	var/list/door_accesses_list = list()
-	var/one_access
-
-	// Stupid shit
-	var/static/allowed_targets = list(/turf, /obj/structure/grille, /obj/structure/window, /obj/structure/lattice, /obj/machinery/door/airlock)
+	/// If this is TRUE, any airlocks that gets built will require only ONE of the checked accesses. If FALSE, it will require ALL of them.
+	var/one_access = TRUE
+	/// Which airlock tab the UI is currently set to display.
+	var/ui_tab = TAB_AIRLOCK_TYPE
+	/// A list of access numbers which have been checked off by the user in the UI.
+	var/list/selected_accesses = list()
+	/// A list of objects that block construction if said object is on the tile the RCD is trying to build on.
+	var/static/list/allowed_targets = list(/turf, /obj/structure/grille, /obj/structure/window, /obj/structure/lattice, /obj/machinery/door/airlock)
```

Comment:
Calling this `allowed_targets` is a little misleading given its things that block construction. Also, are are sure `/turf` is meant to be in here? It seems like it shouldn't.

Reply:
Ah that's actually a mistake on my part with the comment I wrote, I got it backward. It's a list of things you CAN target.